src/Utils/openAuthModal.ts

```diff
@@ -0,0 +1,51 @@
+import * as Schema from "Artsy/Analytics/Schema"
+import { Mediator } from "Artsy/SystemContext"
+import { stringify } from "qs"
+import { data as sd } from "sharify"
+
+interface AuthReason {
+  entity: {
+    id: string
+    name: string
+  }
+  contextModule: Schema.ContextModule | string
+}
+
+export const openAuthModal = (mediator: Mediator, reason: AuthReason) => {
+  if (sd.IS_MOBILE) {
+    openMobileAuth(reason)
+  } else if (mediator) {
+    openDesktopAuth(mediator, reason)
+  } else {
+    window.location.href = "/login"
+  }
+}
+
+const openMobileAuth = ({ entity, contextModule }) => {
+  const params = stringify({
+    action: "follow",
+    contextModule,
+    intent: "follow artist",
+    kind: "artist",
```

Comment:
Right now this seem to only take in account artist entity types. Is that intentional? I ask because the PR title includes following a partner and saving an artwork but that wouldn't be compatible with the hardcoded values.

Reply:
Ahh this is an older commit sorry.